### PR TITLE
📖  General docs cleanup

### PR DESCRIPTION
**Pull request by stmcginnis**
**What this PR does / why we need it**:

Reading through the book for the first time, I took the opportunity to
look at it with fresh eyes and find things that were missing or needed
an update.

This addresses various grammer, spelling, and linking issues found
throughout the book. There are some mdlint updates that were included
where they were very obvious, but that was not the scope of this effort.
If we want to go with full mdlinting, then that should be its own PR and
also include a job to make sure it's enforced.

### CODE REVIEW

File: docs/book/src/developer/e2e.md [hunk 1]
```diff
@@ -58,25 +58,25 @@ Using the config file it is possible to:
...
 
 An [example E2E config file] can be found here.
 
 <aside class="note">
 
 <h1>Deprecated E2E config file format</h1>
 
-The [Cluster API test framework] includes also a [deprecated E2E config file] implementation,
-that was used before the introduction of clusterctl. This might be removed in future releases 
+The [Cluster API test framework] also includes a [deprecated E2E config file] implementation,
```
Comment (fabriziopandini) [inline docs/book/src/developer/e2e.md:70]: AFAIK this has been removed in v1alpha4, so we can get rid of this

File: docs/book/src/reference/glossary.md [hunk 2]
```diff
@@ -5,7 +5,7 @@
 # A
 ---
 
-### Add-ons
```
Comment (fabriziopandini) [inline docs/book/src/reference/glossary.md:8]: I don't have historical context here, but might be we are using h3 intentionally in order to keep the page layout a little bit more concise

File: go.mod [hunk 3]
```diff
@@ -25,6 +25,7 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.7.0
 	go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489
+	golang.org/dl v0.0.0-20210315215437-3c153da08ce0 // indirect
```
Comment (fabriziopandini) [inline go.mod:28]: This seems unrelated to the current PR 🤔

**Review by fabriziopandini — COMMENTED** [review]
/ok-to-test
@stmcginnis thanks for this PR, really appreciated.
Only few nits from a fist pass. I will check preview after it is generated

File: docs/book/src/developer/e2e.md (hunk 1 above)
Comment (stmcginnis) [reply docs/book/src/developer/e2e.md:70]: Great, I will remote this section.

File: docs/book/src/reference/glossary.md (hunk 2 above)
Comment (stmcginnis) [reply docs/book/src/reference/glossary.md:8]: This is one of the things mdlint complains about - going from h1 to h3. But really no reason we can't for formatting reasons, especially since we aren't generating any TOCs based off of this. I will switch these back.

File: go.mod (hunk 3 above)
Comment (stmcginnis) [reply go.mod:28]: Hmm, yeah. I think that must have snuck in by running `make serve-book`. So it may be needed but just omitted before. I will remove it, run that make target again, and see if it gets pulled back in.

Comment (stmcginnis) [reply go.mod:28]: Nope, it didn't. Not sure where that came from, but I will drop it from this patch.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Comment by vincepri** [comment]
Thank you for the great cleanup 😊

**Comment by stmcginnis** [comment]
/retest

Failures do not look doc related.

**Comment by stmcginnis** [comment]
/retest

Seeing this on multiple PRs:

```
E0322 19:09:30.814316   12119 machineset_controller.go:159] controller-runtime/manager/controller/machineset "msg"="Failed to reconcile MachineSet" "error"="failed to retrieve InfrastructureMachineTemplate external object \"md-test\"/\"md-template\": context canceled" "name"="md-wglmb-6dc69c8f65" "namespace"="md-test" "reconciler group"="cluster.x-k8s.io" "reconciler kind"="MachineSet"
```

Doesn't look related.
